test/integration/email_confirmation_test.rb

```diff
@@ -67,4 +67,54 @@ def request_confirmation_mail(email)
 
     assert page.has_content? "Sign out"
   end
+
+  test "requesting confirmation mail with webauthn enabled" do
+    create_webauthn_credential
+
+    request_confirmation_mail @user.email
+
+    link = last_email_link
+    assert_not_nil link
+    visit link
+
+    assert page.has_content? "Multi-factor authentication"
+    assert page.has_content? "Security Device"
+
+    WebAuthn::AuthenticatorAssertionResponse.any_instance.stubs(:verify).returns true
+
+    click_on "Authenticate with security device"
+
+    find(:css, ".header__popup-link").click
+    assert page.has_content?("SIGN OUT")
+
+    @authenticator.remove!
+  end
+
+  def create_webauthn_credential
```

Comment:
Echoing Jenny's suggestion about a helper file. I think this same method is reused in 3 different files (these two and the sign in test) now, so would be helpful to extract

Reply:
Moved this to `test_helper.rb`